### PR TITLE
tests: sweep extended tests for referenced to origin

### PR DESCRIPTION
**Pull request by mfojtik**
Summary of changes:

* Move away from internal clientsets (which should allow to delete some in origin)
* Switch to external API in tests
* Move some image helpers around
* Remove internal client helpers from util

/cc @deads2k 
/cc @soltysh

### CODE REVIEW

File: test/extended/router/scoped.go [hunk 1]
```diff
@@ -155,18 +151,22 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
...
 			e2e.Logf("Selected: %#v, All: %#v", ingress, r.Status.Ingress)
 			o.Expect(ingress).NotTo(o.BeNil())
 			o.Expect(ingress.Host).To(o.Equal(fmt.Sprintf(pattern, "route-1", ns)))
-			external := routev1.RouteIngress{}
-			err = routev1conversions.Convert_route_RouteIngress_To_v1_RouteIngress(ingress, &external, nil)
-			o.Expect(err).NotTo(o.HaveOccurred())
-			status, condition := routedisplayhelpers.IngressConditionStatus(&external, routev1.RouteAdmitted)
-			o.Expect(status).To(o.Equal(corev1.ConditionTrue))
-			o.Expect(condition.LastTransitionTime).NotTo(o.BeNil())
+
```
Comment (mfojtik) [inline test/extended/router/scoped.go]: @squeed @deads2k i'm not sure why is this in extended test or what is the purpose of this test?

Comment (mfojtik) [reply test/extended/router/scoped.go]: i guess if we want this we can import the conversions from openshift-apiserver once we do the move

Comment (deads2k) [reply test/extended/router/scoped.go]: > i guess if we want this we can import the conversions from openshift-apiserver once we do the move

Let's call that plan never.

File: test/extended/router/scoped.go [hunk 2]
```diff
@@ -155,18 +151,22 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
...
 			o.Expect(err).NotTo(o.HaveOccurred())
 			ingress := ingressForName(r, "test-override")
 			e2e.Logf("Selected: %#v, All: %#v", ingress, r.Status.Ingress)
 			o.Expect(ingress).NotTo(o.BeNil())
 			o.Expect(ingress.Host).To(o.Equal(fmt.Sprintf(pattern, "route-1", ns)))
-			external := routev1.RouteIngress{}
-			err = routev1conversions.Convert_route_RouteIngress_To_v1_RouteIngress(ingress, &external, nil)
-			o.Expect(err).NotTo(o.HaveOccurred())
-			status, condition := routedisplayhelpers.IngressConditionStatus(&external, routev1.RouteAdmitted)
-			o.Expect(status).To(o.Equal(corev1.ConditionTrue))
```
Comment (deads2k) [inline test/extended/router/scoped.go]: This check here has value.  Pull the condition and check it.

File: test/extended/router/scoped.go [hunk 3]
```diff
@@ -155,18 +151,22 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
...
-			o.Expect(condition.LastTransitionTime).NotTo(o.BeNil())
```
Comment (deads2k) [inline test/extended/router/scoped.go]: and this one.

File: test/extended/router/scoped.go [hunk 4]
```diff
@@ -223,17 +223,20 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router]", func() {
...
+			r, err := oc.RouteClient().RouteV1().Routes(ns).Get("route-override-domain-2", metav1.GetOptions{})
 			o.Expect(err).NotTo(o.HaveOccurred())
 			ingress := ingressForName(r, "test-override-domains")
 			o.Expect(ingress).NotTo(o.BeNil())
 			o.Expect(ingress.Host).To(o.Equal(fmt.Sprintf(pattern, "route-override-domain-2", ns)))
-			external := routev1.RouteIngress{}
-			err = routev1conversions.Convert_route_RouteIngress_To_v1_RouteIngress(ingress, &external, nil)
-			o.Expect(err).NotTo(o.HaveOccurred())
-			status, condition := routedisplayhelpers.IngressConditionStatus(&external, routev1.RouteAdmitted)
-			o.Expect(status).To(o.Equal(corev1.ConditionTrue))
```
Comment (deads2k) [inline test/extended/router/scoped.go]: likewise here.

File: test/extended/util/image_helpers.go [hunk 5]
```diff
@@ -18,8 +23,65 @@ func DumpAndReturnTagging(tags []string) ([]string, error) {
 	return hexIDs, nil
 }
 
+// ParseImageStreamImageName splits a string into its name component and ID component, and returns an error
+// if the string is not in the right form.
+func ParseImageStreamImageName(input string) (name string, id string, err error) {
```
Comment (deads2k) [inline test/extended/util/image_helpers.go]: moved to library-go.

File: test/extended/util/image_helpers.go [hunk 6]
```diff
@@ -18,8 +23,65 @@ func DumpAndReturnTagging(tags []string) ([]string, error) {
...
+		}
+		result = append(result, tagEventList[i].Items...)
+	}
+	return result
+}
+
+// ImageWithMetadata mutates the given image. It parses raw DockerImageManifest data stored in the image and
+// fills its DockerImageMetadata and other fields.
+// Copied from github.com/openshift/image-registry/pkg/origin-common/util/util.go
+func ImageWithMetadataOrDie(image *imagev1.Image) {
```
Comment (deads2k) [inline test/extended/util/image_helpers.go]: we moved this to `oc`, but if we have another need, move to librar-go

Comment (mfojtik) [reply test/extended/util/image_helpers.go]: i was not sure if e2e test justifies move to library-go

File: test/extended/cluster/cl.go [hunk 7]
```diff
@@ -252,7 +251,7 @@ func newProject(nsName string) *projectapi.Project {
 		ObjectMeta: metav1.ObjectMeta{
 			Name: nsName,
 			Annotations: map[string]string{
-				oapi.OpenShiftDisplayName: nsName,
+				"openshift.io/display-name": nsName,
```
Comment (soltysh) [inline test/extended/cluster/cl.go]: We have annotations.OpenShiftDisplayName from openshift/api/annotations just for that.

File: test/extended/images/hardprune.go [hunk 8]
```diff
@@ -177,42 +178,44 @@ var _ = g.Describe("[Feature:ImagePrune][registry][Serial][Suite:openshift/regis
...
 		o.Expect(err).NotTo(o.HaveOccurred())
 		deleted, err = RunHardPrune(oc, dryRun)
 		o.Expect(err).NotTo(o.HaveOccurred())
+		exutil.ImageWithMetadataOrDie(imgs[childImg1])
 		expectedDeletions = mergeOrSetExpectedDeletions(expectedDeletions,
 			&RegistryStorageFiles{
 				/* TODO: reenable once we delete layer links as well
 				LayerLinks: RepoLinks{oc.Namespace()+"/c": []string{
-					imgs[childImg1].DockerImageMetadata.ID,
+					imgs[childImg1].DockerImageMetadata.Object.(*docker10.DockerImage).ID,
```
Comment (soltysh) [inline test/extended/images/hardprune.go]: Nasty :stuck_out_tongue_closed_eyes:

Comment (mfojtik) [reply test/extended/images/hardprune.go]: pre-existing

/cc @adambkaplan @dmage 

I hope you guys have BZ that track this.

**Comment by mfojtik** [comment]
Updated to use library-go helpers.
/hold cancel

File: test/extended/images/hardprune.go (hunk 8 above)
Comment (adambkaplan) [reply test/extended/images/hardprune.go]: If we didn't have one before, we do now: https://bugzilla.redhat.com/show_bug.cgi?id=1718878

**Comment by squeed** [comment]
cc @ironcladlou, who owns the router / Route api

**Comment by mfojtik** [comment]
@squeed not needed anymore, sorry for noise... i fixed it myself :-)

**Comment by mfojtik** [comment]
Retagging for LGTM (fixed conflicts and scheme and bug in image helper).

**Comment by soltysh** [comment]
Superseded by https://github.com/openshift/origin/pull/23127
/close

**Pull request closed without merging**